### PR TITLE
MLPAB-2128 Make provide certificate tolerate failures

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: internal/notify/client.go [hunk 1]
```diff
@@ -208,3 +246,12 @@ func (c *Client) do(req *http.Request) (response, error) {
 
 	return r, nil
 }
+
+func (c *Client) makeReference(to string, email Email) string {
```
Comment (hawx) [inline internal/notify/client.go:250]: I don't think there is any email we'd want to send to the same person multiple times, right?

File: internal/notify/client.go [hunk 2]
```diff
@@ -165,10 +185,33 @@ func (c *Client) SendActorSMS(ctx context.Context, to, lpaUID string, sms SMS) e
...
+	}
+
+	resp, err := c.do(req)
+	if err != nil {
+		return false, err
+	}
+
+	for _, notification := range resp.Notifications {
+		if (notification.Status == "sending" || notification.Status == "delivered") &&
+			notification.CreatedAt.After(c.now().Add(-allowResendAfter)) {
```
Comment (hawx) [inline internal/notify/client.go:201]: I'm kind of scared of setting a blanket policy on preventing resends (which would be 7 days based on what you can query in notify), so I'm choosing a 10 minute window as that will prevent problems with people resubmitting the form

Comment (acsauk) [reply internal/notify/client.go:197]: Sensible 👍

File: internal/page/certificateprovider/provide_certificate.go [hunk 3]
```diff
@@ -67,12 +71,13 @@ func ProvideCertificate(
...
-				}
 
-				if err := lpaStoreClient.SendCertificateProvider(r.Context(), certificateProvider, lpa); err != nil {
-					return err
+				if lpa.CertificateProvider.SignedAt.IsZero() {
+					if err := lpaStoreClient.SendCertificateProvider(r.Context(), certificateProvider, lpa); err != nil {
+						return err
+					}
+				} else {
+					certificateProvider.Certificate.Agreed = lpa.CertificateProvider.SignedAt
```
[the remark this answers is not in the document]
Comment (acsauk) [reply internal/page/certificateprovider/provide_certificate.go:80]: Maybe that's a sign we align our term for signing to be `SignedAt`?

**Review by acsauk — APPROVED** [review]
:shipit: 

The comment on aligning terms probably applies in a number of places so could be good to look at where we differ from the LPA store way of naming things to see if it can be aligned.

File: internal/page/certificateprovider/provide_certificate.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (hawx) [reply internal/page/certificateprovider/provide_certificate.go:80]: yes, and I'd love to remove the `.Certificate` wrapper

**Pull request merged**